actionpack/lib/action_controller/template_assertions.rb

```diff
@@ -102,6 +102,11 @@ def reset_template_assertion
     #   # assert that the "_customer" partial was rendered with a specific object
     #   assert_template partial: '_customer', locals: { customer: @customer }
     def assert_template(options = {}, message = nil)
+      ActiveSupport::Deprecation.warn(
```

Comment:
We should raise a NoMethodError and point to the gem so we can remove all the implementation